exporter/signalfxexporter/exporter.go

```diff
@@ -175,47 +182,70 @@ func newGzipPool() sync.Pool {
 	}}
 }
 
-func newEventExporter(config *Config, logger *zap.Logger) (*signalfxExporter, error) {
+func newEventExporter(config *Config, createSettings exporter.CreateSettings) (*signalfxExporter, error) {
 	if config == nil {
 		return nil, errors.New("nil config")
 	}
 
-	options, err := config.getOptionsFromConfig()
+	_, err := config.getOptionsFromConfig()
 	if err != nil {
-		return nil,
-			fmt.Errorf("failed to process config: %w", err)
+		return nil, fmt.Errorf("failed to process config: %w", err)
 	}
+	return &signalfxExporter{
+		config:            config,
+		logger:            createSettings.Logger,
+		telemetrySettings: createSettings.TelemetrySettings,
+	}, nil
 
-	headers := buildHeaders(config)
+}
 
-	transport := http.DefaultTransport.(*http.Transport).Clone()
-	transport.MaxIdleConns = config.MaxConnections
-	transport.MaxIdleConnsPerHost = config.MaxConnections
-	transport.IdleConnTimeout = 30 * time.Second
+func (se *signalfxExporter) startLogs(_ context.Context, host component.Host) error {
+	options, err := se.config.getOptionsFromConfig()
+	if err != nil {
+		return fmt.Errorf("failed to process config: %w", err)
+	}
 
-	ingestTLSCfg, err := config.IngestTLSSettings.LoadTLSConfig()
+	client, err := se.createClient(host)
 	if err != nil {
-		return nil, fmt.Errorf("could not load Ingest TLS config: %w", err)
+		return err
 	}
-	transport.TLSClientConfig = ingestTLSCfg
 
 	eventClient := &sfxEventClient{
 		sfxClientBase: sfxClientBase{
 			ingestURL: options.ingestURL,
-			headers:   headers,
-			client: &http.Client{
-				Timeout:   config.Timeout,
-				Transport: transport,
-			},
-			zippers: newGzipPool(),
+			headers:   buildHeaders(se.config),
+			client:    client,
+			zippers:   newGzipPool(),
 		},
-		logger:                 logger,
-		accessTokenPassthrough: config.AccessTokenPassthrough,
+		logger:                 se.logger,
+		accessTokenPassthrough: se.config.AccessTokenPassthrough,
 	}
 
-	return &signalfxExporter{
-		pushLogsData: eventClient.pushLogsData,
-	}, nil
+	se.pushLogsData = eventClient.pushLogsData
+	return nil
+}
+
+func (se *signalfxExporter) createClient(host component.Host) (*http.Client, error) {
+
+	// we want to control how headers are set, overriding user headers with our passthrough.
+	se.config.HTTPClientSettings.Headers = nil
+	se.config.HTTPClientSettings.TLSSetting = se.config.IngestTLSSettings
+
+	if se.config.HTTPClientSettings.MaxIdleConns == nil {
+		se.config.HTTPClientSettings.MaxIdleConns = &se.config.MaxConnections
+	}
+	if se.config.HTTPClientSettings.MaxIdleConnsPerHost == nil {
+		se.config.HTTPClientSettings.MaxIdleConnsPerHost = &se.config.MaxConnections
+	}
+	if se.config.HTTPClientSettings.MaxConnsPerHost == nil {
+		se.config.HTTPClientSettings.MaxConnsPerHost = &se.config.MaxConnections
+	}
```

Comment:
Why is this needed? We didn't touch set `MaxConnsPerHost` before, right?

Reply:
I assumed that was needed and missing before. I'll remove this addition.